backend/internal/auth/sessionmanager.go

```diff
@@ -16,6 +21,17 @@ var (
 	ErrSessionDNE = errors.New("the session does not exist")
 )
 
+// Package specific contextKey type
+type contextKey string
+
+// newContextKey generates a key for storing and accessing values in context.Context.
+// It has a package specific prefix plus additional entropy for added safety, in order
+// to prevent key collisions with other packages.
+func newContextKey() (contextKey, error) {
+	s, err := generateRandomString(8)
+	return contextKey(fmt.Sprintf("teleport-interview-auth.%v", s)), err
```

Comment:
This was based on some research I did a few days ago, and there seems to be some conflicting information online about context keys. In principle it seems that `"teleport-interview-auth"` as it's own package-specific type should be enough to prevent collisions with other packages, this was just me being super paranoid careful. Will simplify.

Reply:
Removed this entirely, `sessionmanager.go` now has the lines
```go
// Package specific contextKey type
type contextKey string

var theContextKey = contextKey("teleport-interview-auth")
```

and then `NewSessionManager` has become
```go
// NewSessionManager creates a new *SessionManager
func NewSessionManager(timeout time.Duration) *SessionManager {
	return &SessionManager{
		store:      make(map[SessionID]Session),
		timeout:    timeout,
		contextKey: theContextKey}
}
```